File: Algorithms/Sorting/Quick Sort/QuickSort.py
array = []

def pivot(array, s, l):
    cc = s - 1
    for i in range(s, l):
        if array[i] <= array[l]:
            cc += 1
            print(i, cc)
            array[cc], array[i] = array[i], array[cc]
            
    array[cc + 1], array[l] = array[l], array[cc+1]
    print( array, s, l, cc+1)
    return cc + 1


def quicksort(array, s, l):
    if s<l:
        p = pivot(array, s, l)
        quicksort(array, s, p-1)
        quicksort(array, p+1, l)

while(len(array)<5):
    array.append(int(input("Enter your no: ")))
    

print("Before sorting: ",array)

quicksort(array, 0 , len(array)-1)

print("After sorting: ",array)

File: Algorithms/Sorting/Quick Sort/QuickSort.cpp
#include<iostream>
using namespace std;

int partition(int array[], int s, int l){
    int current = s - 1;
    for(int i=s; i<l; i++){
        if (array[i]<array[l]){
            current++;
            int temp = array[i];
            array[i] = array[current];
            array[current] = temp;
        }
    }
    int temp = array[l];
    array[l] = array[current+1];
    array[current+1] = temp;
    return current+1;
}

void quicksort(int array[], int s, int l){
    if(s<l){
        int pivot = partition(array, s, l);
        quicksort(array, s, pivot-1);
        quicksort(array, pivot+1, l);
    }
}

int main(){
    int array[5] = {0,0,0,0,0};

    for(int i = 0; i<(sizeof(array)/sizeof(array[0])); i++){
        cin >> array[i]; 
    }

    cout << "Before: ";
    for(int i: array){
        cout << i << ',';
    }
    cout << endl;

    quicksort(array, 0 , sizeof(array)/sizeof(array[0]) - 1);

    cout << "After: ";
    for(int i: array){
        cout << i << ',';
    }
    cout << endl;

    return 0;
}